ros2run/ros2run/api/__init__.py

```diff
@@ -49,11 +49,21 @@ def get_executable_paths(*, node, package_name):
     return executable_paths
 
 
-def get_executable_path(*, node, package_name, executable_name):
-    paths = get_executable_paths(node=node, package_name=package_name)
-    paths2base = {
-        p: os.path.basename(p) for p in paths
-        if os.path.basename(p) == executable_name}
+def get_executable_path(*, package_name, executable_name):
+    paths = get_executable_paths(package_name=package_name)
+    paths2base = {}
+    for p in paths:
+        basename = os.path.basename(p)
+        if basename == executable_name:
+            # pick exact match
+            paths2base[p] = basename
+        elif sys.platform == 'win32':
```

Comment:
No, if there is an exact match it will select that one (line 57-59). Only if not filenames with additional extensions according to `PATHEXT` are being considered.

So in your case if you call `<pkgname> talker` it will execute the file named `talker`.